.genx/templates/grid.hbs

```diff
@@ -1,9 +1,20 @@
-<zero-grid-pro ${ref('tile1grid1')}>
+<zero-grid-pro
+>
   <grid-pro-genesis-datasource
     resource-name="{{config.resourceName}}"
     {{#if config.snapshot}}
     isSnapshot="{{config.snapshot}}"
     {{/if}}
+      {{#if config.fdc3RowClick}}
+    :deferredGridOptions="${() => {
+            return {
+                onRowClicked: async (e: RowSelectedEvent) => {
+                    const payload = e.data
+                    sendMessageOnChannel('{{config.fdc3RowClick.channel}}', '{{config.fdc3RowClick.type}}', payload)
+                }
+            }
+        }}"
```

Comment:
If we add a util similar to this (basically a getter for the event handler function):

```ts
export const sendEventOnChannel = (channelName: string, type: string) => {
    return async (e: any) => {
        // check for ag-grid-specific events, fall back to standard events
        const payload = e.data?.payload || e.detail;
        sendMessageOnChannel(channelName, type, payload);
    }
}
```

we could simplify the configuration block to just:

```ts
:deferredGridOptions="${() => ({
  onRowClicked: sendEventOnChannel('{{config.fdc3RowClick.channel}}', '{{config.fdc3RowClick.type}}')
})}"
```

Reply:
that's very grid specific so will add another method